### PR TITLE
Feature: Make separate networks ippo build from shared networks ippo.

### LINKED ISSUE #707: [FEATURE] Make ippo with separate and shared networks more similar
### Please describe the purpose of the feature. Is it related to a problem?
Get ippo with separated and shared networks to use more of the same components. 

### Describe the solution you'd like
Try to get both systems to use the same parameter client and parameter server components. Furthermore, maybe make one of the systems use the other system as a basis. E.g. the shared networks IPPO uses the independent network IPPO as a basis, but updates the trainer components.

### PR DESCRIPTION
**Pull request by DriesSmit**
## What?
Make the separate networks ippo system build from the shared networks ippo.
## Why?
This shows how components can be used to upgrade a system.
## How?
Changed the ippo system file.
## Extra
The network sizes are now also updated to be an exponent of 2.
Close #707

### CODE REVIEW

**Review by RuanJohn — APPROVED** [review]
Thanks @DriesSmit, this look really good! :muscle:

File: mava/specs.py [hunk 1]
```diff
@@ -170,3 +170,11 @@ def get(self) -> Dict[str, Any]:
             _description_
         """
         return self.__dict__
+
+    def set(self, name: str, component: Any) -> None:
+        """_summary_
```
Comment (RuanJohn) [inline mava/specs.py:175]: Can we please add a descriptive dosctring here?

**Review by RuanJohn — CHANGES_REQUESTED** [review]
Thank you @DriesSmit! It looks :fire: Just one minor request from my side.

File: mava/components/jax/building/parameter_client.py [hunk 2]
```diff
@@ -238,7 +238,7 @@ def name() -> str:
 ###################
 
 
-class ExecutorParameterClientSeparateNetworks(BaseParameterClient):
+class ExecutorParameterClientSeparateNetworks(ExecutorParameterClient):
```
Comment (AsadJeewa) [inline mava/components/jax/building/parameter_client.py:241]: I get why you did this but the code in this class is exactly the same as ExecutorParameterClient besides when we create the networks so if we extend this we should reuse code or extend base

**Review by AsadJeewa — COMMENTED** [review]
Thanks for this @DriesSmit 
Small comment

File: examples/jax/debugging/simple_spread/feedforward/decentralised/run_ippo.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def main(_: Any) -> None:
     # Networks.
     def network_factory(*args: Any, **kwargs: Any) -> Any:
         return ippo.make_default_networks(  # type: ignore
-            policy_layer_sizes=(254, 254, 254),
```
Comment (AsadJeewa) [inline examples/jax/debugging/simple_spread/feedforward/decentralised/run_ippo.py:67]: Out of interest, would be good to know why this was set to54. Obviously we want a power of 2 but weird that it was set to this

Comment (DriesSmit) [reply examples/jax/debugging/simple_spread/feedforward/decentralised/run_ippo.py:67]: I think it was just an arbitrary choice :)

File: mava/specs.py (hunk 1 above)
Comment (DriesSmit) [reply mava/specs.py]: Fixed :+1:

File: mava/components/jax/building/parameter_client.py (hunk 2 above)
Comment (DriesSmit) [reply mava/components/jax/building/parameter_client.py:241]: Nice catch @AsadJeewa :smile: I updated the code. Does it look better now?

File: mava/specs.py (hunk 1 above)
Comment (RuanJohn) [reply mava/specs.py]: Thank you :)

**Review by RuanJohn — DISMISSED** [review]
Thank you @DriesSmit :muscle:

**Review by AsadJeewa — APPROVED** [review]
Looks good. Thanks, @DriesSmit

File: mava/components/jax/building/parameter_client.py (hunk 2 above)
Comment (AsadJeewa) [reply mava/components/jax/building/parameter_client.py:241]: I am happy

**Review by RuanJohn — APPROVED** [review]
Thank you @DriesSmit :muscle:

**Pull request merged**